Sources/Examples/ReflectionService/Documentation.docc/ReflectionServiceTutorial.md

```diff
@@ -0,0 +1,177 @@
+It receives the paths to the files containing the reflection data of the proto files 
+describing the services of the sqerver that we want to be discoverable through reflection
+and the version of the reflection service.
+
+
+### Generating the reflection data
+The server from this example uses the `GreeterProvider` and the `EchoProvider`,
+besides the `ReflectionService`.
+  The associated proto files are located at 
+ `Sources/Examples/HelloWorld/Model/helloworld.proto`, and 
+ `Sources/Examples/Echo/Model/echo.proto` respectively.
+
+ In order to generate the reflection data for the
+ `helloworld.proto`, you can run the following command:
+
+```sh
+$ protoc Sources/Examples/HelloWorld/Model/helloworld.proto \
+    --proto_path=Sources/Examples/HelloWorld/Model \
+    --plugin=./.build/debug/protoc-gen-grpc-swift \
+    --grpc-swift_opt=Client=false,Server=false,ReflectionData=true \
+    --grpc-swift_out=Sources/Examples/ReflectionService/Generated
+```
+
+Let's break the command down:
+- The first argument passed to `protoc` is the path 
+to the `.proto` file to generate reflection data
+for: [`Sources/Examples/HelloWorld/Model/helloworld.proto`][helloworld-proto].
+- The `proto_path` flag is the path to search for imports: 
+`Sources/Examples/HelloWorld/Model`.
+- The plugin we are using is `./.build/debug/protoc-gen-grpc-swift`.
+- The `grpc-swift_opt` options are passed to the `protoc-gen-grpc-swift` plugin. 
+To generate only the reflection data set `Client=false,Server=false,ReflectionData=true`.
+- The `grpc-swift_out` flag is used to set the path of the directory
+where the generated file will be located: `Sources/Examples/ReflectionService/Generated`.
+
+The commands for generating the reflection data for the `Echo` and `Reflection` services are similar.
+
+Depending on the version of [gRPCurl][grpcurl] you are using you might need to use the v1alpha proto instead.
+Beginning with [gRPCurl v1.8.8][grpcurl-v188] it uses the [v1][v1] reflection. Earlier versions use [v1alpha][v1alpha]
+reflection.
+
+ ### Instantiating the Reflection Service 
+
+To instantiate the `ReflectionService` you need to pass the file paths of
+the generated reflection data and the version to use, in our case `.v1`.
+
+```swift
+let paths = [
+  "Sources/Examples/ReflectionService/Generated/helloworld.grpc.reflection.txt", 
+  "Sources/Examples/ReflectionService/Generated/echo.grpc.reflection.txt", 
+]
+
+let reflectionServiceProvider = try ReflectionService(
+  serializedFileDescriptorProtoFilePaths: paths, 
+  version: .v1
+)
+```
+
+### Running the Server
+
+In our example the server isn't configured with TLS and listens on localhost port 1234.
+The following code configures and starts the server:
+
+```swift
+let server = try await Server.insecure(group: group)
+  .withServiceProviders([reflectionService, GreeterProvider(), EchoProvider()])
+  .bind(host: "localhost", port: self.port)
```

Comment:
Hmm the example uses `self.port` because it uses an argument parser and defaults to 1234, this looks a bit misleading with the text above. Maybe we should just put 1234 here so it's in line with the text above?